index.js

```diff
@@ -27,4 +27,78 @@ module.exports = function (emitter, event) {
     };
 
     return promise;
-};
+}
+
+function _isThenable (obj) {
+    return typeof obj.then === 'function';
+}
+
+function _isEmitter (obj) {
+    return typeof obj.once === 'function';
+}
+
+function _parseRaceArgs (args) {
+    var n = args.length;
+    var i = 0;
+    var promises = [];
+
+    while (i < n) {
+        if (_isThenable(args[i])) {
+            promises.push(args[i]);
+            i += 1;
+            continue;
+        }
+
+        if (_isEmitter(args[i]) && i + 1 < n && typeof args[i+1] === 'string') {
+            promises.push(promisifyEvent(args[i], args[i+1]));
+            i += 2;
+            continue;
+        }
+
+        throw new Error('Wrong arguments');
+    }
+
+    return promises;
+}
+
+function _getPromisesCancellations (promises) {
+    return promises.map(function (promise){
+        if (typeof promise.cancel !== 'function')
+            return void 0;
+
+        return promise.cancel();
+    });
+}
+
+function _wrapRacePromises (promises) {
+    return promises.map(function (promise, index) {
+        return promise
+            .then(function (result){
+                return { index: index, result: result };
+            })
+            .catch(function (error){
+                return { index: index, error: error }
+            });
+    });
+}
+
+function raceEvents (args) {
```

Comment:
If we will just cancel all properties after one of them was fullfilled, that one fullfilled promise will be 'canceled' too. Though it does nothing bad in case  of `promisifedEvent`, that doesn't feel right to me in more general case. What do you think about that?